### PR TITLE
Blob gas fee checking

### PR DESCRIPTION
**Pull request by jflo**
fixes: #6391

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java [hunk 1]
```diff
@@ -168,6 +170,18 @@ private ValidationResult<TransactionInvalidReason> validateCostAndFee(
                 "total blob gas %d exceeds max blob gas per block %d",
                 txTotalBlobGas, gasLimitCalculator.currentBlobGasLimit()));
       }
+      if (maybeBlobFee.isEmpty()) {
+        throw new IllegalArgumentException(
+            "blob fee must be provided from blocks containing blobs");
+      } else if (maybeBlobFee.get().compareTo(transaction.getMaxFeePerBlobGas().get()) > 0) {
+        return ValidationResult.invalid(
+            TransactionInvalidReason.GAS_PRICE_BELOW_CURRENT_BASE_FEE,
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java:178]: I think we need a new specific invalid reason for this

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PermissionTransactionValidator.java [hunk 2]
```diff
@@ -43,8 +43,9 @@ public PermissionTransactionValidator(
   public ValidationResult<TransactionInvalidReason> validate(
       final Transaction transaction,
       final Optional<Wei> baseFee,
+      final Optional<Wei> blobBaseFee,
       final TransactionValidationParams transactionValidationParams) {
-    return delegate.validate(transaction, baseFee, transactionValidationParams);
+    return delegate.validate(transaction, baseFee, Optional.empty(), transactionValidationParams);
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PermissionTransactionValidator.java:48]:
```suggestion
    return delegate.validate(transaction, baseFee, blobBaseFee, transactionValidationParams);
```

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/TransactionValidator.java [hunk 3]
```diff
@@ -35,6 +35,7 @@ public interface TransactionValidator {
   ValidationResult<TransactionInvalidReason> validate(
       Transaction transaction,
       Optional<Wei> baseFee,
+      Optional<Wei> blobFee,
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/TransactionValidator.java:38]:
```suggestion
      Optional<Wei> blobBaseFee,
```

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java [hunk 4]
```diff
@@ -300,13 +306,60 @@ public void shouldRejectTransactionWithMaxPriorityFeeGreaterThanMaxFee() {
...
-        validator.validate(transaction, Optional.of(Wei.ONE), transactionValidationParams);
+        validator.validate(
+            transaction, Optional.of(Wei.ONE), Optional.empty(), transactionValidationParams);
     assertThat(validationResult)
         .isEqualTo(ValidationResult.invalid(MAX_PRIORITY_FEE_PER_GAS_EXCEEDS_MAX_FEE_PER_GAS));
     assertThat(validationResult.getErrorMessage())
         .isEqualTo("max priority fee per gas cannot be greater than max fee per gas");
   }
 
+  @Test
```
Comment (fab-10) [inline ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java:317]: Is there already a test that check the opposite, when a tx is valid?

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java [hunk 5]
```diff
@@ -300,13 +306,60 @@ public void shouldRejectTransactionWithMaxPriorityFeeGreaterThanMaxFee() {
...
+  public void shouldRejectTransactionWithMaxBlobPriorityFeeGreaterThanMaxBlobFee() {
```
Comment (fab-10) [inline ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java:318]: not sure, but it seems that should be the name

```suggestion
  public void shouldRejectTransactionWithMaxBlobPriorityFeeSmallerThanBlobBaseFee() {
```

**Review by fab-10 — CHANGES_REQUESTED** [review]
overall looks good, some suggestions

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java (hunk 4 above)
Comment (jflo) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java:317]: yes, `shouldAcceptTransactionWithAtLeastOneBlob` covers this with a tx that overpays.

**Review by fab-10 — APPROVED** [review]
LGTM, will be nice to have a unit test that check the tx is valid even if underpriced when it comes from the txpool

**Pull request merged**